Objects/structseq.c

```diff
@@ -399,6 +400,31 @@ initialize_structseq_dict(PyStructSequence_Desc *desc, PyObject* dict,
     SET_DICT_FROM_SIZE(visible_length_key, desc->n_in_sequence);
     SET_DICT_FROM_SIZE(real_length_key, n_members);
     SET_DICT_FROM_SIZE(unnamed_fields_key, n_unnamed_members);
+
+    // Prepare and set __match_args__
+    PyObject* keys = PyTuple_New(n_members - n_unnamed_members);
+    if (keys == NULL) {
+        return -1;
+    }
+
+    Py_ssize_t i, k;
+    for (i = k = 0; i < n_members; ++i) {
+        if (desc->fields[i].name == PyStructSequence_UnnamedField) {
```

Comment:
I don't think we can handle this in a better way. The other (much worse option IMHO) is generate some missing names for these.